src/main/java/seedu/budgetbaby/logic/statistics/Statistics.java

```diff
@@ -0,0 +1,56 @@
+package seedu.budgetbaby.logic.statistics;
+
+import java.time.YearMonth;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import seedu.budgetbaby.model.BudgetBabyModel;
+import seedu.budgetbaby.model.month.Month;
+
+public class Statistics {
+
+    private final Month currMonth;
+    private BudgetBabyModel model;
+
+    /**
+     * Instantiates the Statistics object by taking in a BudgetBabyModel during initialisation.
+     * @param model BudgetBabyModel object.
+     */
+    public Statistics(BudgetBabyModel model) {
+        this.model = model;
+        currMonth = model.getCurrentDisplayMonth();
+    }
+
+    /**
+     * Checks if a {@code Month} falls within the past 6 months of the currently displayed month.
+     * @param month {@code Month} that is being checked.
+     * @return True if it falls within the past 6 months, and False otherwise.
+     */
+    private boolean withinSixMonths(Month month) {
+        YearMonth curr = month.getMonth();
+        YearMonth end = currMonth.getMonth();
+        YearMonth start = end.minusMonths(6);
+        return (curr.equals(end) || curr.equals(start) || (curr.isBefore(end) && curr.isAfter(start)));
+    }
+
+    private List<Month> getPastMonths() {
+        List<Month> monthList = model.getFilteredMonthList();
+        return monthList.stream().filter(this::withinSixMonths).collect(Collectors.toList());
+    }
+
+    /**
+     * Obtains statistics of past 6 {@code Month}s as a string and returns it.
+     * @return Statistics of past 6 months.
+     */
+    public String getPastMonthStatistics() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM");
+        StringBuilder s = new StringBuilder();
+        for (Month m : getPastMonths()) {
+            s.append(String.format("%s: %.2f/%.2f\n", m.getMonth().format(formatter), m.getTotalExpenses(),
+                    m.getBudget().getAmount()));
+        }
+        return s.toString();
+    }
```

Comment:
Just a thought, is it a good idea to return past month's statistics as a String? Maybe it would be a better idea to return as a `List` or `HashMap`? That way, it will be easier to format and display on the Ui side.

Reply:
Yes, I was just thinking about that too. Looking at the other methods that were implemented in BudgetBabyLogicManager, I realised that it shouldn't be a problem if I returned a List. Thanks for the suggestion!